### PR TITLE
Small UI fixes for seats & destination calendars

### PR DESCRIPTION
**Pull request by joeauyeung**
## What does this PR do?

This PR fixes a number of small UI bugs

When booking new seats, the dates are no longer crossed out
**Before**
image

**After**
image

When the default destination calendar option is selected in `DestinationCalendarSelector`, rather than showing the external id of the calendar it now shows the calendar name and integration
**Before**
image

**After**
image

Add seats badge to event type item
image

Fixes # (issue)

**Environment**: Staging(main branch) / Production

## Type of change

- [x] Chore (refactoring code, technical debt, workflow improvements)

## How should this be tested?

- [ ] Test A
- [ ] Test B

## Checklist

- I haven't read the [contributing guide](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md)
- My code doesn't follow the style guidelines of this project
- I haven't performed a self-review of my own code and corrected any misspellings
- I haven't commented my code, particularly in hard-to-understand areas
- I haven't checked if my PR needs changes to the documentation
- I haven't checked if my changes generate no new warnings
- I haven't added tests that prove my fix is effective or that my feature works
- I haven't checked if new and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by CarinaWolli** [comment]
@joeauyeung I noticed one other small UI issue around seats, that could maybe be fixed in this PR: 
When seats ist enabled, and then you disable it, it still shows the banner 'Seats option doesn't support confirmation requirement'. You then first need to click save with seats disabled and then you can enable 'requires confirmation'

File: apps/web/components/booking/pages/BookingPage.tsx [hunk 1]
```diff
@@ -166,7 +166,8 @@ const BookingPage = ({
           isSuccessBookingPage: true,
           email: bookingForm.getValues("email"),
           eventTypeSlug: eventType.slug,
-          formerTime: booking?.startTime.toString(),
+          ...(!eventType.seatsPerTimeSlot &&
```
Comment (CarinaWolli) [inline apps/web/components/booking/pages/BookingPage.tsx:169]: now we also don't show the former date crossed out when rescheduling a seat

**Review by CarinaWolli — CHANGES_REQUESTED** [review]
Please address my comment below. Other than that, everything looks good to me

File: apps/web/components/booking/pages/BookingPage.tsx [hunk 2]
```diff
@@ -166,7 +166,7 @@ const BookingPage = ({
           isSuccessBookingPage: true,
           email: bookingForm.getValues("email"),
           eventTypeSlug: eventType.slug,
-          formerTime: booking?.startTime.toString(),
+          ...(rescheduleUid && booking?.startTime && { formerTime: booking.startTime.toString() }),
```
Comment (joeauyeung) [inline apps/web/components/booking/pages/BookingPage.tsx:169]: Now shows former time when rescheduling a seat

File: apps/web/components/eventtype/EventAdvancedTab.tsx [hunk 3]
```diff
@@ -74,7 +74,7 @@ export const EventAdvancedTab = ({ eventType, team }: Pick<EventTypeSetupProps,
   const [selectedCustomInputModalOpen, setSelectedCustomInputModalOpen] = useState(false);
   const [requiresConfirmation, setRequiresConfirmation] = useState(eventType.requiresConfirmation);
   const placeholderHashedLink = `${CAL_URL}/d/${hashedUrl}/${eventType.slug}`;
-  const seatsEnabled = formMethods.getValues("seatsPerTimeSlotEnabled");
+  const seatsEnabled = formMethods.watch("seatsPerTimeSlotEnabled");
```
Comment (joeauyeung) [inline apps/web/components/eventtype/EventAdvancedTab.tsx:77]: Actively watches if seats are enabled, so the requires confirmation prompt will update without saving.

**Review by joeauyeung — COMMENTED** [review]
@CarinaWolli

**Review by CarinaWolli — APPROVED** [review]
LGTM 👏🏻

**Pull request merged**